quadmmwave_icp_core:
  ros__parameters:
    input_topic_0: /radar/scan_0
    input_topic_1: /radar/scan_1
    input_topic_2: /radar/scan_2
    input_topic_3: /radar/scan_3
    imu_topic: /imu/data
    base_frame: base_link
    odom_frame: odom
    sensor0_extrinsics: [0.00,  0.00,  0.00, 0.0, 0.0, 0.0]
    sensor1_extrinsics: [-0.03, 0.00,  0.07, 0.0, -0.785398, 0.0]
    sensor2_extrinsics: [ 0.03, 0.00, -0.07, 0.0,  0.785398, 0.0]
    sensor3_extrinsics: [-0.02, 0.07,  0.00, 0.0,  0.0,      0.785398]
    x_rot_deg: 90.0
    voxel_min: 0.06
    voxel_max: 0.16
    sor_k: 50
    sor_std: 1.3
    pass_radius_min: 0.25
    pass_radius_max: 14.0
    pass_z_min: -3.0
    pass_z_max: 3.0
    pass_intensity_min: 6.0
    icp_max_iterations: 60
    icp_fitness_epsilon: 1.0e-5
    icp_transformation_epsilon: 1.0e-9
    icp_adapt_corr_min: 0.15
    icp_adapt_corr_max: 0.60
    ransac_thresh: 0.08
    use_reciprocal_corr: true
    scale_correction: 1.0
    keyframe_trans_thresh: 0.80
    keyframe_rot_thresh_deg: 10.0
    use_ndt_fallback: true

quadmmwave_mapping:
  ros__parameters:
    aligned_topic: /quadmmwave/points/aligned
    pose_topic:    /quadmmwave/pose
    path_topic:    /quadmmwave/path
    odom_topic:    /quadmmwave/odom
    accum_topic:   /quadmmwave/points/accum
    map_topic:     /quadmmwave/map
    map_frame:  map
    odom_frame: odom
    base_frame: base_link
    vis_scale: 1.0
    yaw_deg: 0.0
    x_offset: 0.0
    y_offset: 0.0
    z_offset: 0.0
    chunk_distance_m: 1.8
    resolution: 0.08
    limit_cells: 12000
    logodds_occ: 2.0
    logodds_free: -0.7
    logodds_min: -4.0
    logodds_max: 4.0
    prob_threshold: 0.5